pilot/test/xdstest/grpc.go

```diff
@@ -0,0 +1,74 @@
+package xdstest
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"istio.io/pkg/log"
+)
+
+func safeSleep(ctx context.Context, t time.Duration) {
+	select {
+	case <-time.After(t):
+	case <-ctx.Done():
+	}
+}
+
+type slowClientStream struct {
+	grpc.ClientStream
+	recv, send time.Duration
+}
+
+func (w *slowClientStream) RecvMsg(m interface{}) error {
+	if w.recv > 0 {
```

Comment:
Where is w.recv updated? Does it need a mutex protection?

Reply:
Its not mutated, just set once at construction